### PR TITLE
[fix] link custom setting

### PR DESCRIPTION
**Pull request by MD931**
Hi Thomas,

This PR fixes the redirection links for custom settings to redirect to the custom setting setup page instead object manager page.

image

instead

image

The same goes for object list records for custom setting.

Thank you :)

### CODE REVIEW

File: addon/popup.js [hunk 1]
```diff
@@ -986,9 +988,11 @@ class AllDataSelection extends React.PureComponent {
   /**
    * Optimistically generate lightning setup uri for the provided object api name.
    */
-  getObjectSetupLink(sobjectName, durableId) {
+  getObjectSetupLink(sobjectName, durableId, isCustomSetting) {
     if (sobjectName.endsWith("__mdt")) {
       return this.getCustomMetadataLink(durableId);
+    } else if(isCustomSetting) {
+      return "https://" + this.props.sfHost + "/lightning/setup/CustomSettings/page?address=%2F"+ durableId +"?setupid=CustomSettings";
```
Comment (tprouvot) [inline addon/popup.js:995]:
```suggestion
    } else if (isCustomSetting) {
      return "https://" + this.props.sfHost + "/lightning/setup/CustomSettings/page?address=%2F" + durableId + "?setupid=CustomSettings";
```

File: addon/popup.js [hunk 2]
```diff
@@ -998,18 +1002,22 @@ class AllDataSelection extends React.PureComponent {
   getCustomMetadataLink(durableId) {
     return "https://" + this.props.sfHost + "/lightning/setup/CustomMetadata/page?address=%2F" + durableId + "%3Fsetupid%3DCustomMetadata";
   }
-  getObjectFieldsSetupLink(sobjectName, durableId) {
+  getObjectFieldsSetupLink(sobjectName, durableId, isCustomSetting) {
     if (sobjectName.endsWith("__mdt")) {
       return this.getCustomMetadataLink(durableId);
+    } else if(isCustomSetting) {
+      return "https://" + this.props.sfHost + "/lightning/setup/CustomSettings/page?address=%2F"+ durableId +"?setupid=CustomSettings";
```
Comment (tprouvot) [inline addon/popup.js:1009]:
```suggestion
    } else if (isCustomSetting) {
      return "https://" + this.props.sfHost + "/lightning/setup/CustomSettings/page?address=%2F" + durableId + "?setupid=CustomSettings";

```

File: addon/popup.js [hunk 3]
```diff
@@ -998,18 +1002,22 @@ class AllDataSelection extends React.PureComponent {
...
     } else {
       return "https://" + this.props.sfHost + "/lightning/setup/ObjectManager/" + sobjectName + "/FieldsAndRelationships/view";
     }
   }
-  getObjectListLink(sobjectName, keyPrefix) {
+  getObjectListLink(sobjectName, keyPrefix, isCustomSetting) {
     if (sobjectName.endsWith("__mdt")) {
       return "https://" + this.props.sfHost + "/lightning/setup/CustomMetadata/page?address=%2F" + keyPrefix;
+    } else if(isCustomSetting) {
+      return "https://" + this.props.sfHost + "/lightning/setup/CustomSettings/page?address=%2Fsetup%2Fui%2FlistCustomSettingsData.apexp?id=" + keyPrefix;
```
Comment (tprouvot) [inline addon/popup.js:1020]:
```suggestion
    } else if (isCustomSetting) {
      return "https://" + this.props.sfHost + "/lightning/setup/CustomSettings/page?address=%2Fsetup%2Fui%2FlistCustomSettingsData.apexp?id=" + keyPrefix;

```

**Review by tprouvot — CHANGES_REQUESTED** [review]
Hi Mouloud thanks for this PR !
Just a few suggestions from ESLint on the use of spaces :

image

**Comment by MD931** [comment]
Hello Thomas,

Thank you for your feedback and sorry for not having seen ESlint.

I just directly committed your suggestions.

Thanks.

**Pull request merged**